### PR TITLE
[Governance] Add correlationID in body for error responses

### PR DESCRIPTION
**Pull request by VivekVinushanth**
### Proposed changes in this pull request
This PR adds **traceId** in the body of error response so that the client can keep track of the error and this sorted out.

### CODE REVIEW

**Comment by VivekVinushanth** [comment]
Fixes 
- https://github.com/wso2-enterprise/asgardeo-product/issues/8135

File: components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java [hunk 1]
```diff
@@ -76,7 +76,10 @@ public Response getMe() {
             userAttributes = Utils.getUserInformationService().getRetainedUserInformation(username, userStoreDomain,
                     tenantId);
         } catch (UserExportException e) {
-            return Response.serverError().entity(e.getMessage()).build();
+            ErrorDTO errorDTO = new ErrorDTO();
+            errorDTO.setRef(Utils.getCorrelation());
+            errorDTO.setMessage(e.getMessage());
+            return Response.serverError().entity(errorDTO).build();
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java:82]: Are you changing the response structure here?

File: components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java [hunk 2]
```diff
@@ -187,6 +189,31 @@ public static NotFoundException buildNotFoundException(String description, Strin
...
+     * Get correlation id of current thread
+     *
+     * @return correlation-id
+     */
+    public static String getCorrelation() {
+        String ref;
+        if (isCorrelationIDPresent()) {
+            ref = MDC.get(Constants.CORRELATION_ID_MDC).toString();
+        } else {
+            ref = UUID.randomUUID().toString();
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java:211]: don't we need to add the generated id to MDC if not present?

File: components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java (hunk 1 above)
Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java:82]: Yes. The response body will have a new field **traceId**

File: components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java (hunk 2 above)
Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java:211]: Fixed with 54e80535d2c691cd16bbc38ee1548f83cab98dda

**Review by senthalan — APPROVED** [review]

**Pull request merged**